src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/AddRolesToExistingUserFuncTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.reform.userprofileapi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.restassured.RestAssured;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import net.serenitybdd.junit.spring.integration.SpringIntegrationSerenityRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.userprofileapi.client.*;
+import uk.gov.hmcts.reform.userprofileapi.config.TestConfigProperties;
+
+@RunWith(SpringIntegrationSerenityRunner.class)
+public class AddRolesToExistingUserFuncTest extends AbstractFunctional {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AddRolesToExistingUserFuncTest.class);
+
+    @Autowired
+    protected TestConfigProperties configProperties;
+
+    private IdamClient idamClient;
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = targetInstance;
+        RestAssured.useRelaxedHTTPSValidation();
+        idamClient = new IdamClient(configProperties);
+    }
+
+    @Test
+    public void should_update_user_profile_with_roles_successfully() throws Exception {
+
+
+        CreateUserProfileData data = createUserProfileData();
+        List<String> roles = new ArrayList<>();
+        roles.add(puiUserManager);
+        String email = idamClient.createUser(roles);
+
+        data.setEmail(email);
+        createUserProfile(data, HttpStatus.CREATED);
+
+        RoleName role1 = new RoleName(puiCaseManager);
+        Set<RoleName> rolesName = new HashSet<>();
+        rolesName.add(role1);
+        UpdateUserProfileData userRProfileData = new UpdateUserProfileData();
+        userRProfileData.setRolesAdd(rolesName);
+
+        GetUserProfileResponse resource =
+                testRequestHandler.sendGet(
+                        requestUri + "?email=" + email.toLowerCase(),
+                        GetUserProfileResponse.class
+                );
+
+        LOG.info("before addroles call");
+        UserProfileRolesResponse resource1 =
+                testRequestHandler.sendPut(
+                        userRProfileData,
+                            HttpStatus.OK,
+                           requestUri + "/" + resource.getIdamId(), UserProfileRolesResponse.class);
+
+        LOG.info("after addroles call" + resource1);
+
+        GetUserProfileWithRolesResponse resource2 =
+                testRequestHandler.sendGet(
+                        "/v1/userprofile/" + resource.getIdamId() + "/roles",
+                        GetUserProfileWithRolesResponse.class
+                );
+        LOG.info("Roles addroles call" + resource2);
+        assertThat(resource2.getRoles().size()).isNotNull();
+        assertThat(resource2.getRoles().size()).isEqualTo(3);

```

Comment:
Along with size it is also better to check exact roles which got added newly from GetUserProfileWithRolesResponse

Reply:
done